src/Workspaces/CSharp/Portable/CodeGeneration/ConstructorGenerator.cs

```diff
@@ -50,7 +50,7 @@ private static MemberDeclarationSyntax LastConstructorOrField(SyntaxList<MemberD
             var reusableSyntax = GetReuseableSyntaxNodeForSymbol<ConstructorDeclarationSyntax>(constructor, options);
             if (reusableSyntax != null)
             {
-                return reusableSyntax;
+                return RemoveLeadingDirectiveTrivia(reusableSyntax);
```

Comment:
so i am curious why you don't always have GetReuseableSyntaxNodeForSymbol do this.  can you clarify?

Reply:
https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces/CodeGeneration/FieldGenerator.cs,85
Here, the caller of GetReuseableSyntaxNodeForSymbol is trying to get a reusable `VariableDeclarator' node, and checking the parent of return value of the method.
So If we let GetReuseableSyntaxNodeForSymbol to do the work, then the parent of resuable `VariableDeclartor` is null.

Reply:
Another possible way is let the FieldGenerator to use a special version of `GetReuseableSyntaxNodeForSymbol` (which doesn't modify its parent)
Does this fix sound better?

Reply:
Yes. Though I'd say that the flag should be about trivia.

Reply:
I'm also curious why FieldGenerator needs the parent.

Reply:
Because the field symbol is binded to VariableDeclarator, and for code like
```
class Bar
{
   public int i = 10, j = 100, $$k = 1000;
}
```
If the symbol is 'k', FieldGenerator is going to fetch VariableDeclator for `k` and, 
and find its parent FieldDeclaration to generate `public int k = 1000;`

Reply:
but i don't get why it needs the parnet FieldDeclaration.  why can't it figure everything out from the IFieldSymbol it has?

Reply:
Are you saying for the given IFieldSymbol there should be a method checking all its modifier to generate FieldDeclaration instead of reusing the existing FieldDeclaration?

Reply:
Yeah.  i'm basically saying that i have no idea why we need to walk up to the parent FieldDeclaration.  Why is the IFieldSymbol+VarDecl not enough for us to meet the needs of whatever feature is using this.

Reply:
OK, so here is all the usages of impl of this function which needs special treatment
1. VariableDeclarator. (Need to find its parent FieldDeclaration, but might not be needed)
2. https://github.com/dotnet/roslyn/pull/55450 addresses a VB side problem
3. This PR, which reomves the leading directive trivia

Do you feel it would be OK to put it in ISematicFacts? Maybe we should call it `TryGetReusableSyntaxNodeForSymbol`?

Reply:
@CyrusNajmabadi  In case this thread is forgotten after the weekend : )

Reply:
I'm not sure i understand your question.  First, i find the 'reuse syntax' concept somewhat suspect in the first place.  Why are we asking to generate syntax from a symbol if we already have the syntax in teh first place?  It just doesn't really make any sense, and i wish i knew why we added that.  

Second, even if we want to reuse the syntax, then teh syntax for a single field-symbol is the var-decl, not the field-decl.  So we should just reuse that, and then synthesize anything else we need.

If some feature genuinely needs to reuse the entire field-decl i would want to know why, and i would say that it's likely it needs to be written specially to handle that.  For example, if we're generating a field-decl from a field-sym, and the existing field-decl has two var-decls, then if we reuse the field-decl we'll continue to get two var-decls.  It's all very strange and i think represents behavior we should nto be trying to maintain.